typespec-tests/src/main/java/com/cadl/specialheaders/models/Resource.java

```diff
@@ -41,6 +44,27 @@ public final class Resource implements JsonSerializable<Resource> {
     @Generated
     private final String type;
 
+    @Generated
+    private boolean jsonMergePatch;
+
+    /**
+     * Stores updated model property, the value is property name, not serialized name.
+     */
+    @Generated
+    private final Set<String> updatedProperties = new HashSet<>();
+
+    @Generated
+    void serializeAsJsonMergePatch(boolean jsonMergePatch) {
+        this.jsonMergePatch = jsonMergePatch;
+    }
+
+    static {
```

Comment:
Currently we cannot add `@Generated` annotation to static code block, so partial updatemay treat this static code block as customized code. We may need to provide a whitelist for partial update supported elements: member properties, member methods, interface implemented types. Other elements change will be ignored by partial-update. I've tracked this in this issue: https://github.com/Azure/autorest.java/issues/2468.